debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/Filters.java

```diff
@@ -57,7 +57,16 @@ else if (collectionExcludeList != null && !collectionExcludeList.trim().isEmpty(
             collectionFilter = (id) -> true;
         }
         Predicate<CollectionId> isNotBuiltIn = this::isNotBuiltIn;
-        this.collectionFilter = isNotBuiltIn.and(collectionFilter);
+        Predicate<CollectionId> finalCollectionFilter = isNotBuiltIn.and(collectionFilter);
+        String signalDataCollection = config.getString(MongoDbConnectorConfig.SIGNAL_DATA_COLLECTION);
+        if (signalDataCollection != null) {
+            CollectionId signalDataCollectionId = CollectionId.parse("UNUSED", signalDataCollection);
```

Comment:
Could you please extend the `CollectionId` contract so no dummy value is needed to be passed?

Reply:
What exactly are you thinking on? The CollectionId constructor expects three non null strings, provided that assertions are enabled.

I just took as example this line in the MongoDbIncrementalSnapshotChangeEventSource constructor:
        this.signallingCollectionId = connectorConfig.getSignalingDataCollectionId() == null ? null
                : CollectionId.parse("UNUSED", connectorConfig.getSignalingDataCollectionId());

Do you want to fix also this place?

Reply:
@ggaborg I see, then please create a follow-up Jira to get rid off both instances in a separate PR later.

Reply:
@jpechane, created https://issues.redhat.com/browse/DBZ-5575 "Get rid of dummy value used to parse mongodb CollectionId"